### PR TITLE
refactor: use aliases instead of meta

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes [UNIFY-869](https://cypress-io.atlassian.net/browse/UNIFY-869)

### User facing changelog

Probably more stable testing with vite.

### Additional details

With the ambition to pregenerate vitejs tests this refactoring was necessary.
It brings the build philosophy closer to other rollup/vite plugins and makes it buildable.

Changes:
- instead of using `import.meta.env.__cypress_supportPath`and `__cypress_originAutUrl` use strongly typed import aliases `cypress:config`
- instead of loading the specs purely dynamically, resolve all the specs in a plugin generated specLoaders file and load it with the alias `cypress:spec-loaders`. If the current spec is not pre-loaded, use a purely dynamic, but that should be the exception
- resolve the supportfile statically on the server and use the `cypress:support-path` alias to access it
- move the `startServer.ts/start()` code to `index.ts` to simplify architecture
- rename the file to `resolveServerConfig.ts` since it now only contains the config resolution

### CODE REVIEW

File: npm/vite-dev-server/src/makeCypressPlugin.ts [hunk 1]
```diff
@@ -48,40 +45,56 @@ export const makeCypressPlugin = (
...
+    load (id) {
+      if (id === 'cypress:spec-loaders') {
+        return `export default {\n${specs.map((s) => {
+          return `${JSON.stringify(s.relative)}:()=>import(${JSON.stringify(s.absolute)})`
+        }).join(',\n')}\n}`
+      }
+
+      if (id === 'cypress:config') {
+        return `
+export const supportPath = ${JSON.stringify(!!supportFilePath)}
```
Comment (lmiller1990) [inline npm/vite-dev-server/src/makeCypressPlugin.ts:92]: Won't this always eval to `export const supportPath = false`? Should it be

```js
${JSON.stringify(!!supportFilePath ? supportFilePath : false)}
```

**Review by lmiller1990 — COMMENTED** [review]
Left one comment. Also it should be easy to add a `system-test` for this behavior (perhaps one for `supportFile: '/somewhere'` and `supportFile: false`? You could look at `system-tests/tests/run-ct_spec.js`. I don't think we have any system-tests using Vite so this would be quite nice to have. What do you think?

File: npm/vite-dev-server/src/makeCypressPlugin.ts (hunk 1 above)
Comment (elevatebart) [reply npm/vite-dev-server/src/makeCypressPlugin.ts:92]: Agreed, what I meant was `hasSupportFile`

**Comment by elevatebart** [comment]
Adding a system test, is a very good idea, I will add it before I merge this request.

**Comment by lmiller1990** [comment]
@elevatebart system tests are available for CT, look here: https://github.com/cypress-io/cypress/blob/master/system-tests/test/run_ct_spec.js

**Comment by elevatebart** [comment]
I cannot understand how I missed that. Probably a case of the Mondays. Will make sure it's tested tomorrow and ping you then.

**Comment by elevatebart** [comment]
System test added

**Comment by elevatebart** [comment]
system test added [system-tests/test/vite_example_spec.ts](https://github.com/cypress-io/cypress/pull/19566/files#diff-40d5f7f21e3565e06607e34e8e0baa29bbea360f87e4546d2e1da02f7f22d284)

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**